src/frontend/src/components/DateField/index.tsx

```diff
@@ -14,19 +14,41 @@ interface Props {
   formik: FormikContextType<any>;
   helperText?: any;
   onChange?: (d: ChangeEvent) => void;
+  shouldShowEditedCellsAsMarked?: boolean;
 }
 
+export type valueType = string | boolean | NamedGisaidLocation | undefined;
+
 export default function DateField({
   fieldKey,
   formik,
   helperText,
   onChange,
+  shouldShowEditedCellsAsMarked,
 }: Props): JSX.Element {
-  const { handleChange, handleBlur, values, touched, errors } = formik;
+  const { handleChange, handleBlur, values, touched, errors, initialValues } =
+    formik;
+  const [isBackgroundColorShown, setBackgroundColorShown] =
+    useState<boolean>(false);
+  const [changedValue, setChangedValue] = useState<valueType>(undefined);
+  const [initialValue, setInitialValue] = useState<valueType>(undefined);
+
+  useEffect(() => {
+    setChangedValue(values[fieldKey]);
+    setInitialValue(initialValues[fieldKey]);
+  }, [fieldKey, initialValues, values]);
+
+  useEffect(() => {
+    if (initialValue !== changedValue && shouldShowEditedCellsAsMarked) {
+      setBackgroundColorShown(true);
+    } else {
+      setBackgroundColorShown(false);
+    }
+  }, [initialValue, changedValue, shouldShowEditedCellsAsMarked]);
```

Comment:
these hooks are reused between the `DateField`, `FreeTextField`, and `LocationField`, is there a more dry way to do this? (@mmmmmmaya @vincent-czi )

Reply:
you can turn it into a custom hook. https://reactjs.org/docs/hooks-custom.html

Reply:
Does the `initialValue` change over time? I would think that would be constant for the entirety of time the Edit modal is open, because it's what the metadata started as before editing process begins. So you could get rid of that state, possibly move it up to props, but if you need the overall `initialValues` object at this level, just have it pull as a constant on every render or `useMemo` it.

Similarly -- and I don't know enough to be sure, haven't read PR or have a ton of context -- it seems like this doesn't need to be done with a useEffect or have a state for the background color. Just do a comparison like that if-conditional, and pass the boolean as a prop to the `StyledBlah` component.

As for underlying question though, I don't really know how doable it is to extract cleanly and generalize to all the various Fields. This comparison will probably have to happen inside each of these Fields since that's where the comparison can have access to the `currentValue`, right? I guess you could use the formik at the parent level and compare up there? Not sure that's really so much better. But if you shrink the usage of state and effect hooks, the boilerplate aspect will probably be less cumbersome.

Reply:
@vincent-czi yeah you're right the initialValue does not change over time, so i could get rid of that state. the UseEffect i think is necessary though for when the user changes the value (changedValue) that would trigger the color change

Reply:
I'm still not sure why the useEffect has to happen though? That seems like it could just be determined on every render which would happen with `changedValue` changing. If `changedValue` changes, it will kick off a re-render, and you can just have a comparison in "normal" code without a useEffect hook.

Reply:
yeah i guess this is a little overkill